### ISSUE TITLE
Qusetion about training speed、test and valid set

### ISSUE THREAD

**Issue by lyqqhgzj** [AlexeyAB/darknet#4846]
hi @AlexeyAB 
I have some questions to consult you.

1. My data set has 7851 images and only two categories. Average training time lasts about five minutes. Is that slow? The program doesn't seem to be using the GPU(GPU usage 37MiB / 2048MiB).

2. I followed the steps（How to train (to detect your custom objects):） as if I didn't create the test set and valid set. Will the program automatically divide this two parts？
my   obj.data   file :
classes= 2
train  = data/train.txt
valid  = data/test.txt
names = data/obj.names
backup = backup/

3. The program always comes out with three epochs at once. The results get worse each epoch. It takes most of time for the first epoch, and the results come out quickly for the last two epochs.
After 30 epochs results and three consecutive epochs:
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 82 Avg (IOU: 0.509385, GIOU: 0.503955), Class: 0.510964, Obj: 0.477977, No Obj: 0.559059, .5R: 0.500000, .75R: 0.000000, count: 2, loss = 355.380981, class_loss = 354.809357, iou_loss = 0.571625
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 94 Avg (IOU: 0.286392, GIOU: 0.158159), Class: 0.453461, Obj: 0.490958, No Obj: 0.486175, .5R: 0.076923, .75R: 0.000000, count: 13, loss = 1099.743774, class_loss = 1086.548828, iou_loss = 13.194946
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 106 Avg (IOU: 0.230466, GIOU: 0.119039), Class: 0.480330, Obj: 0.548931, No Obj: 0.550219, .5R: 0.031579, .75R: 0.010526, count: 95, loss = 5652.409668, class_loss = 5507.151367, iou_loss = 145.258301

loss = 5652.409668 ！！！  Big Number！！！
I don't know if the program is running correctly.
I would appreciate it if you could reply as soon as possible！！！

**Comment by AlexeyAB** [comment]
You should compile Darknet with GPU=1 in the Makefile.

**Comment by lyqqhgzj** [comment]
Anyhow, thank you for your advice.
I compiled it. But it is the same speed as before. Is there something wrong with the previous step?

As IMAGE（How to compile on Windows (using CMake-GUI)） shows, when I use CMake-GUI, I can't configure and generate if two files have the same path. 
Then I make a change.
Where is the source code: D:/PycharmProjects/darknet-master (a copy of darknet-master)
Where to build the binaries: D:/VSWorkSpace/darknet-master
The bebug file was generated in darknet-master. The output of VS2017 is 6 successes 0 failures 1 skip.   ???  1 skip  ???    Is it ok?

I am reconfiguring the above environment with a new computer. One more thing, using the CMake-GUI method, whether the installation order for vs2017 also must precede cuda?

**Comment by lyqqhgzj** [comment]
I did it all over again, and now the speed is about one epoch per minute. Until then, GPU usage is still 37Mib.

**Comment by AlexeyAB** [comment]
If you use Cmake, then CUDA and CUDNN should be enabled.

![image](https://user-images.githubusercontent.com/4096485/74339158-8b9d1100-4db4-11ea-9ac5-7aa091f6acd0.png)

**Comment by lyqqhgzj** [comment]
Before Configure, I click them. Before Generate, I click them. And before Open Project, I click them. There is no change. However, the good news is now using my brother's computer training, there are 6 gigabytes of gpu(usage 287MiB/6144MiB), a minute will appear three v3. But, my training runs at 4,000 iterations would be about 7.7 days, if my calculations are correct.
（4000 iterations, 7851 images, batch = 64, about 60s get 3 v3(mes = ..............), about 857s loaded 64 images）
using commond nvidia-smi -i 0 
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 432.00       Driver Version: 432.00       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+==========|
|   0  GeForce RTX 2060   WDDM  | 00000000:09:00.0  On |                  N/A |
|  0%   24C    P8    21W / 175W |    287MiB /  6144MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=================================================================|
|    0      1248    C+G   Insufficient Permissions                   N/A      |
|    0      4416    C+G   C:\Windows\explorer.exe                    N/A      |
|    0      7196    C+G   ...t_cw5n1h2txyewy\ShellExperienceHost.exe N/A      |
|    0      7624    C+G   ...dows.Cortana_cw5n1h2txyewy\SearchUI.exe N/A      |
|    0      7644    C+G   ...\2017\Enterprise\Common7\IDE\devenv.exe N/A      |
|    0      8576    C+G   ...hell.Experiences.TextInput.InputApp.exe N/A      |
+-----------------------------------------------------------------------------+

I'm not sure which one is the name of the training process. I noticed that the first one said "Insufficient Permissions." I tried to run with powershell. Or is there no training program.

**Comment by AlexeyAB** [comment]
Training with GPU=1 CUDNN=1 should be much more faster.

**Comment by lyqqhgzj** [comment]
I have compiled the Makefile and set GPU=1,CUDNN=1. Nothings changed. What else do you think is wrong?
Makefile:
GPU=1
CUDNN=1
CUDNN_HALF=0
OPENCV=0
AVX=0
OPENMP=0
LIBSO=0
ZED_CAMERA=0

**Comment by lyqqhgzj** [comment]
sorry, I didn't click Grouped and Advance. In the Ungrouped Entries, there are 4 selections(
CUDA_VERBOSE_BUILD, 
MANUALLY_EXPORT_TRACK_OPTFLOW,
SELECT_OPENCV_MODULES,
USE_INTEGRATED_LIBS).
Which ones should I choose？

**Comment by lyqqhgzj** [comment]
I reinstalled cuda lite. No matter what I do, I can't use the GPU training program. My brain exploded. OMG!!!

**Comment by AlexeyAB** [comment]
1. Do you use Windows or Linux?
2. Do you use Make, or CMake, or vcpkg, or legacy way for Darknet compilation?
3. What CUDA, cuDNN and OpenCV version did you install?
4. Show output of command `nvcc --version`
5. Show such screenshot of starting training: 
![image](https://user-images.githubusercontent.com/4096485/74420630-d1151900-4e5c-11ea-8fd0-c059f8177d9e.png)

6. Show such screenshot of Cmake-GUI
![image](https://user-images.githubusercontent.com/4096485/74339158-8b9d1100-4db4-11ea-9ac5-7aa091f6acd0.png)

**Comment by lyqqhgzj** [comment]
1. windows10
2. CMake-GUI  version 3.16
3. cuda_10.1.243_426.00_win10  cudnn-10.1-windows10-x64-v7.6.5.32    opencv-3.4.9-vc14_vc15
4. nvcc --version
![Q %S)BYV)AW$C(L5~8LKBKK](https://user-images.githubusercontent.com/58103660/74423552-70e99b80-4e8b-11ea-8858-d47bae3a22c1.png)
5. powershell 
![5COWA93JSSC$PAMCOEB}MF2](https://user-images.githubusercontent.com/58103660/74423717-abebcf00-4e8b-11ea-8df3-59d4de775f2a.png)
6. cmake
![(69 {20DZAJ}$LAE7BB8}KV](https://user-images.githubusercontent.com/58103660/74423862-e6556c00-4e8b-11ea-8c9b-c7b93a2b2f7a.png)

**Comment by AlexeyAB** [comment]
> I reinstalled **cuda lite.** 

What is the CUDA Lite?
What MSVS version do you use?
Install CUDA https://developer.nvidia.com/cuda-downloads

Show such screenshot of MSVS
![image](https://user-images.githubusercontent.com/4096485/74424516-25bb9280-4e63-11ea-8c4c-d8ea214b5441.png)

**Comment by lyqqhgzj** [comment]
Translate the results of the software. This installation has a must first install VS.
![)VSI}PAQ1FOU}$S{Q22VBQO](https://user-images.githubusercontent.com/58103660/74425261-5d8bff80-4e8e-11ea-8911-29a881265c1d.png)

vs2017
![3DY(L$F96RAT0A6A7Z``KQW](https://user-images.githubusercontent.com/58103660/74425078-13a31980-4e8e-11ea-96d1-a2c143c5ec93.png)

**Comment by AlexeyAB** [comment]
Cmake doesn't see your CUDA. https://github.com/AlexeyAB/darknet/issues/4846#issuecomment-585649877

@cenit Hi, Do you know what could be the reason?

**Comment by lyqqhgzj** [comment]
after open project
![UHTHHED2JGE{%`30S$9ANJ0](https://user-images.githubusercontent.com/58103660/74426441-56fe8780-4e90-11ea-9df8-1ec84585faf2.png)

**Comment by lyqqhgzj** [comment]
How is self_driving.data file generated? And did you add the entire darknet file to the system environment variable? Otherwise, why does darknet.exe run in x64?

> 1. Do you use Windows or Linux?
> 2. Do you use Make, or CMake, or vcpkg, or legacy way for Darknet compilation?
> 3. What CUDA, cuDNN and OpenCV version did you install?
> 4. Show output of command `nvcc --version`
> 5. Show such screenshot of starting training:
>    ![image](https://user-images.githubusercontent.com/4096485/74420630-d1151900-4e5c-11ea-8fd0-c059f8177d9e.png)
> 6. Show such screenshot of Cmake-GUI
>    ![image](https://user-images.githubusercontent.com/4096485/74339158-8b9d1100-4db4-11ea-9ac5-7aa091f6acd0.png)

**Comment by cenit** [comment]
can you please launch the script .\build.ps1 which is in the darknet root folder from a powershell and post the build logs here? it might contain useful informations (and maybe even build correctly darknet in case it can self-heal your config :) )

**Comment by lyqqhgzj** [comment]
what commond should I enter in powershell?
![WLGIFIN{$ZUX1BDYJOE6645](https://user-images.githubusercontent.com/58103660/74463686-2a1c9580-4ecd-11ea-9725-8ae9ba1a206a.png)

**Comment by lyqqhgzj** [comment]
only find build file 
![SDG9~)S~4A6MW4~}2PYYNVN](https://user-images.githubusercontent.com/58103660/74463898-88e20f00-4ecd-11ea-814a-e5cbb95cf6d5.png)

**Comment by lyqqhgzj** [comment]
I launch this scrip, but after running, it flashes back directly.

**Comment by lyqqhgzj** [comment]
It's 2pm in China now, I think I should go to bed. Thank you for your suggestion. You can just write your idea on it. I'll give it a try.

**Comment by cenit** [comment]
the `security exception` is a good hint.
You have to first "unlock" your powershell, which is locked down as of now.
The command is

`Set-ExecutionPolicy unrestricted`

after that, running `.\build.ps1` should work

**Comment by lyqqhgzj** [comment]
.\build.ps1
![1T1Q5BSW$UPYM1LDEGAUCHV](https://user-images.githubusercontent.com/58103660/74500980-aa6de580-4f23-11ea-974f-ac655bff7538.png)

The release folder generated after opening open project is directly under darkne-master, and the command I run is also switched to the release folder.
![4ONH0J` )Y%@8H3SA@J PJ](https://user-images.githubusercontent.com/58103660/74501173-3bdd5780-4f24-11ea-836d-f95f155340db.png)

**Comment by cenit** [comment]
Can you please clone darknet from scratch in a new folder and relaunch .\build.ps1, in order to start from a clean environment? Then please post the log

**Comment by lyqqhgzj** [comment]
sorry, I may not get you mean correctly. I uncompressed the downloaded darknet-master and put it in newfold without making CMake, and then executed the command you said.
![B$4%)T1ZY$%)_1XP54$ ZCR](https://user-images.githubusercontent.com/58103660/74506078-c24d6580-4f33-11ea-94d8-50201291d2b8.png)

**Comment by lyqqhgzj** [comment]
When using CMake-GUI configure, 
![Z1O )UFP@~9}E1Q6{SJRKI](https://user-images.githubusercontent.com/58103660/74508382-e7dd6d80-4f39-11ea-8410-627175efb0d2.png)

I've added the path to the system environment variable
![6`B521@AP1D1IJ{EJGJQ1$S](https://user-images.githubusercontent.com/58103660/74508392-ee6be500-4f39-11ea-808f-8ca6f170ee99.png)

**Comment by lyqqhgzj** [comment]
hi @AlexeyAB ,
It's better to reinstall the system than to find errors slowly. The result was a one-time success. Many of the previous failures were due to problems with CMAKE and the slow download of github files in China. After reviewing a lot of data, the configuration was finally successful. Now it only takes 3 seconds to train an epoch. Finally, thanks to the author. Then I'll list some of the configuration procedures for more people to learn.
I don't know why I still can't generate darknet.exe in x64. All my files are under the Release folder in darknet-master, so the training program is also conducted in this environment. Is that all right?
![VH(CHPR}3UEWGN{V`UAM2TT](https://user-images.githubusercontent.com/58103660/74698341-68e97d00-5238-11ea-8b31-69d3c2c75bb0.png)

**Comment by AlexeyAB** [comment]
>  Is that all right?

Yes.

**Comment by lyqqhgzj** [comment]
Provide some blogs for others to read.
[https://blog.csdn.net/weijifen000/article/details/93377143](url)
[https://blog.csdn.net/liqiancao/article/details/100034821](url)
My computer environment：
vs2017
cmake-3.16.4-win64-x64.msi
opencv3.4.5
opencv_contribe3.4.5
cuda10
cudnn7.5
Once again, I sincerely thank the author.
